demos/python_demos/whiteboard_inpainting_demo/utils/network_wrappers.py

```diff
@@ -0,0 +1,123 @@
+                             'got {}x{}.'.format(self.w, self.h, im_data.shape[2], im_data.shape[1]))
+        im_data = np.pad(im_data, ((0, 0),
+                                   (0, self.h - im_data.shape[1]),
+                                   (0, self.w - im_data.shape[2])),
+                         mode='constant', constant_values=0).reshape(1, self.c, self.h, self.w)
+        im_info = im_info.reshape(1, *im_info.shape)
+        output = self.net.infer(dict(im_data=im_data, im_info=im_info))
+
+        classes = output['classes']
+        valid_detections_mask = classes > 0
+        classes = classes[valid_detections_mask]
+        boxes = output['boxes'][valid_detections_mask]
+        scores = output['scores'][valid_detections_mask]
+        masks = output['raw_masks'][valid_detections_mask]
+        return boxes, classes, scores, np.full(len(classes), 0, dtype=np.int32), masks
+
+    def get_detections(self, frames, return_cropped_masks=False, only_class_person=True):
+        outputs = []
+        for frame in frames:
+            data_batch = self._preprocess(frame)
+            im_data = data_batch['im_data']
+            im_info = data_batch['im_info']
+            meta = data_batch['meta']
+
+            boxes, classes, scores, _, masks = self.forward(im_data, im_info)
+            scores, classes, boxes, masks = postprocess(scores, classes, boxes, masks,
+                                                        im_h=meta['original_size'][0],
+                                                        im_w=meta['original_size'][1],
+                                                        im_scale_y=meta['processed_size'][0] / meta['original_size'][0],
+                                                        im_scale_x=meta['processed_size'][1] / meta['original_size'][1],
+                                                        full_image_masks=True, encode_masks=False,
+                                                        confidence_threshold=self.confidence)
+            frame_output = []
+            for i in range(len(scores)):
+                if only_class_person and classes[i] != 1:
+                    continue
+                bbox = [int(value) for value in boxes[i]]
+                if return_cropped_masks:
+                    left, top, right, bottom = bbox
+                    mask = masks[i][top:bottom, left:right]
+                else:
+                    mask = masks[i]
+                frame_output.append([bbox, scores[i], mask])
+            outputs.append(frame_output)
+        return outputs
+
+
+class SemanticSegmentation(IEModel):
+    def __init__(self, ie, model_path, conf=.6, device='CPU', ext_path='', max_reqs=1):
+        super().__init__(ie, model_path, conf, device, ext_path, max_reqs)
+
+    def get_detections(self, frames, only_class_person=True):
+        outputs = []
+        for frame in frames:
+            out_h, out_w = frame.shape[:-1]
+            res = self.forward(frame)
```

Comment:
I think there should be resize like in `_preprocess()`

Reply:
Resize is used in the `_preprocess` method implemented in the base class.